### PR TITLE
azurerm_machine_learning_workspace - encryption block

### PR DESCRIPTION
**Pull request by xuzhang3**
Support   `Customer-managed key`  data  encryption

AccTest:

```log
TF_ACC=1 go test -v ./internal/services/machinelearning -run=TestAccMachineLearningWorkspaceDataSource_basic -timeout 60m -ldflags="-X=github.com/hashicorp/terraform-provider-azure
TF_ACC=1 go test -v ./internal/services/machinelearning -run=TestAccMachineLearningWorkspaceDataSource_basic -timeout 60m -ldflags="-X=github.com/hashicorp/terraform-provider-azure
TF_ACC=1 go test -v ./internal/services/machinelearning -run=TestAccMachineLearningWorkspaceDataSource_basic -timeout 60m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccMachineLearningWorkspaceDataSource_basic
=== PAUSE TestAccMachineLearningWorkspaceDataSource_basic
=== CONT  TestAccMachineLearningWorkspaceDataSource_basic
--- PASS: TestAccMachineLearningWorkspaceDataSource_basic (2451.67s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/machinelearning       2454.984s



TF_ACC=1 go test -v ./internal/services/machinelearning -run=TestAccMachineLearningWorkspace -timeout 60m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccMachineLearningWorkspaceDataSource_basic
=== PAUSE TestAccMachineLearningWorkspaceDataSource_basic
=== RUN   TestAccMachineLearningWorkspace_basic
=== PAUSE TestAccMachineLearningWorkspace_basic
=== RUN   TestAccMachineLearningWorkspace_requiresImport
=== PAUSE TestAccMachineLearningWorkspace_requiresImport
=== RUN   TestAccMachineLearningWorkspace_basicUpdate
=== PAUSE TestAccMachineLearningWorkspace_basicUpdate
=== RUN   TestAccMachineLearningWorkspace_complete
=== PAUSE TestAccMachineLearningWorkspace_complete
=== RUN   TestAccMachineLearningWorkspace_completeUpdate
=== PAUSE TestAccMachineLearningWorkspace_completeUpdate
=== CONT  TestAccMachineLearningWorkspaceDataSource_basic
=== CONT  TestAccMachineLearningWorkspace_basicUpdate
=== CONT  TestAccMachineLearningWorkspace_basic
=== CONT  TestAccMachineLearningWorkspace_complete
=== CONT  TestAccMachineLearningWorkspace_completeUpdate
=== CONT  TestAccMachineLearningWorkspace_requiresImport
--- PASS: TestAccMachineLearningWorkspace_basic (477.62s)
--- PASS: TestAccMachineLearningWorkspace_basicUpdate (486.27s)
--- PASS: TestAccMachineLearningWorkspace_requiresImport (486.62s)
--- PASS: TestAccMachineLearningWorkspace_completeUpdate (2108.19s)
--- PASS: TestAccMachineLearningWorkspaceDataSource_basic (2377.40s)
--- PASS: TestAccMachineLearningWorkspace_complete (2425.39s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/machinelearning       2429.002s
```

### CODE REVIEW

File: internal/services/machinelearning/machine_learning_workspace_resource.go [hunk 1]
```diff
@@ -141,6 +141,27 @@ func resourceMachineLearningWorkspace() *pluginsdk.Resource {
...
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"key_vault_id": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: keyVaultValidate.VaultID,
+						},
+						"identifier": {
```
Comment (denniseik) [inline internal/services/machinelearning/machine_learning_workspace_resource.go:156]: Can we call this attribute 'key_id' to be consistent with other interfaces? There we use key_uri as we request an Azure resourceId but Terraform handles that better :)

File: website/docs/r/machine_learning_workspace.html.markdown [hunk 2]
```diff
@@ -108,6 +201,14 @@ An `identity` block supports the following:
 
 * `type` - (Required) The Type of Identity which should be used for this Disk Encryption Set. At this time the only possible value is `SystemAssigned`.
 
+---
+
+An `encryption` block supports the following:
+
+* `key_vault_id` - (Required) The ID of the keyVault where the customer owned encryption key is present.
+
+* `identifier` - (Required) The Key Vault URI to access the encryption key.
```
Comment (denniseik) [inline website/docs/r/machine_learning_workspace.html.markdown:210]: Please update based on earlier comment

File: internal/services/machinelearning/machine_learning_workspace_resource.go (hunk 1 above)
Comment (xuzhang3) [reply internal/services/machinelearning/machine_learning_workspace_resource.go]: @denniseik  renamed to `key_id`

**Review by katbyte — APPROVED** [review]
Thanks @xuzhang3 - LGTM 🏗️

**Pull request merged**